src/main/java/net/openhft/chronicle/queue/impl/single/TableDirectoryListing.java

```diff
@@ -31,11 +33,13 @@
             final @NotNull TableStore<?> tableStore,
             final Path queuePath,
             final ToIntFunction<File> fileToCycleFunction,
-            final boolean readOnly) {
+            final boolean readOnly,
```

Comment:
you could drop readOnly as an argument and assert !tableStore.readOnly()

Reply:
Yep, love it